### DISCUSSION TITLE
Documentation for moveit_ros_benchmarks (ROS2 Humble).

### DISCUSSION THREAD

**Discussion by aizmailovs** [moveit/moveit2#2710] [Q&A]
Hi all, 
I have decided to try out the [benchmarking tool](https://github.com/ros-planning/moveit2/tree/e9b214ccb7a1f99732a5489d4d031980cf4c09ff/moveit_ros/benchmarks) to test motion planners. After some work on a package itself, I have made it up and running (thanks to [this](https://discourse.ros.org/t/fixing-moveit2-humble-moveit-ros-benchmarks-package/32048) post and a [git project](https://github.com/sjahr/moveit_resources/tree/d826fc4ea35243e27dd54e04f518e09673b995c6/benchmarking)). However, I do not really understand some of the benchmark attributes, their purpose and measurement units.

To give an example, I do not understand what is the difference between "time", "process time", "final path time". Moreover, I do not really get what is "final smoothness". Also, I cannot really see what are the measurement units (in case of distances and time I can guess it is in SI units, but for smoothness it is a bit trickier).

Is there a documentation available?

**Comment by sjahr** [comment]
@aleksandrsizmailovs Unfortunately, there is not any documentation available at the moment and the attributes need to be cleaned up. I got it working last year and added [a moveit2 tutorial for it](https://moveit.picknik.ai/main/doc/how_to_guides/benchmarking/benchmarking_tutorial.html). Since the attributes are inherited from the [planner arena tool](https://plannerarena.org/) you might find more documentation about them there. Please open a PR with updates for the descriptions if you find something useful or consider refactoring attributes that don't make sense in your opinion
